pkg/apis/pipeline/v1alpha1/gcs_resource_test.go

```diff
@@ -26,7 +26,14 @@ import (
 	corev1 "k8s.io/api/core/v1"
 )
 
-func Test_Invalid_NewStorageResource(t *testing.T) {
+const activateServiceAccountScript = `#!/usr/bin/env bash
```

Comment:
NB: This is copied here because the other version is not exported (and shouldn't be), and tests are in `package v1alpha1_test`.

Is there a preferred way to maintain these invariants together?

Reply:
Should this be in an external package ? (as it's solely related to `gcs`, maybe it should be in the `apis` package at all ?

Reply:
I'd like it not to be exported anywhere at all. But I guess if I have to, a separate package is better.

At least when resources are pluggable all this will leave this repo, right?

Reply:
@ImJasonH that's kinda my thought yeah :angel:

Reply:
+1 for separate package. Seems like the best option for now!

Reply:
In the end I decided to make the resource add a single step with a script that activates the SA then calls `gsutil cp`/`gsutil rsync`, and to just put the full details of the script into each test.

When this gets extensible-ized this will all get cleaner tests will be split out with the code.